src/overlays/actors/ovl_Bg_Ydan_Sp/z_bg_ydan_sp.c

```diff
@@ -63,29 +73,377 @@ static ColliderTrisInit D_808C09A8 = {
         COLSHAPE_TRIS,
     },
     2,
-    D_808C0930,
+    sTrisItemsInit,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Sp/BgYdanSp_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Sp/BgYdanSp_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+void BgYdanSp_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanSp* this = THIS;
+    ColliderTrisElementInit* ti0 = &sTrisItemsInit[0];
+    Vec3f tri[3];
+    s32 i;
+    CollisionHeader* colHeader;
+    ColliderTrisElementInit* ti1 = &sTrisItemsInit[1];
+    f32 cossY;
+    f32 sinsY;
+    f32 cossX;
+    f32 nSinsX;
+
+    colHeader = NULL;
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->isDestroyedSwitchFlag = thisx->params & 0x3F;
+    this->burnSwitchFlag = (thisx->params >> 6) & 0x3F;
+    thisx->params = (thisx->params >> 0xC) & 0xF;
+    DynaPolyActor_Init((DynaPolyActor*)thisx, DPM_PLAYER);
```

Comment:
Many instances of casting `this` or `thisx` are in this file and should be cleaned up, like this one:
```suggestion
    DynaPolyActor_Init(&this->dyna, DPM_PLAYER);
```